### DISCUSSION TITLE
How do I change the date display format on GitHub?

### DISCUSSION THREAD

**Discussion by Mishasama** [community/community#136129] [New to GitHub] [answered]
For non-native speakers, the US display format makes recognizing the date shown quickly difficult and is not conducive to copying and pasting directly to quote the content.
GitHub is supposed to be a global community, not just a US community, right? There's no reason to adopt a US-only standard instead of the internationally agreed ISO standard as default...

e.g. What is Mar? May? Apr? Aug? Jan? Jun? Jul? Totally no idea at all. ㄟ( ▔へ ▔ )ㄏ

Also, want to see the actually time too. #16550

**Comment by AnshulJagotra** [comment, marked as answer]
GitHub currently uses the US date format, which can be confusing for non-native speakers. While there’s no direct option to change the date format on GitHub, here are some workarounds:

Use Browser Extensions: Install extensions like Tampermonkey or Greasemonkey to run custom scripts that convert dates into your preferred format. You can also use libraries like Moment.js within these scripts to handle date and time formatting.

Time Zones and 24-Hour Format: Similar scripts can adjust the time display to show 24-hour format or convert times to your local timezone.

Submit a Feature Request: Consider suggesting this feature to GitHub through their feedback channels. If more users request this, GitHub might consider adding it as an option.

Conversion Tools: Use quick online date converters or scripts to manually convert GitHub dates into a format you’re comfortable with.

These steps can help you better manage date and time displays on GitHub.

**Reply by Mishasama** [reply]
1. Do you know any script can do this work?
2. Where are their feedback channels?

Thanks for your reply.

**Reply by AnshulJagotra** [reply]
1) Yes, you can use a script with extensions like Tampermonkey or Greasemonkey to change the date format on GitHub. Here’s a basic idea:

Install Tampermonkey or Greasemonkey: These are browser extensions that allow you to run custom scripts on websites.

Script Example: You can use JavaScript to modify the date format. A library like Moment.js can help with this. 

2)To submit a feature request or feedback to GitHub:

GitHub Community Forum: You can post suggestions and requests [here](https://github.community/).
GitHub Support: You can contact GitHub support directly through their [support page](https://support.github.com/).

**Reply by Mishasama** [reply]
Sorry, maybe I didn't make myself clear.

1. I don't know how to write JS, I can only just use ready-made scripts imported into TM and run them.
2. This is the community forum, right? Either of the two links you gave will eventually lead here. So, is my question considered feedback?

**Reply by SavinduAbeysooriya** [reply]
There’s still no official setting on GitHub to change the date format globally (for example, to ISO 8601 or a non-US format), but you can achieve this with an easy userscript — no need to write your own from scratch.

Here’s a step-by-step method that works:

Install a userscript manager like Tampermonkey
 (Chrome, Edge) or Violentmonkey
 (Firefox).

Add this ready-made script:

Go to GreasyFork – GitHub Date Format Converter
 (example community script).

It automatically detects GitHub date strings (like “Aug 15, 2024”) and converts them into ISO format (2024-08-15) or your browser’s locale.

Refresh GitHub — all visible dates (commit timestamps, PR dates, discussions) will show in your chosen or local format.

Alternative option:

Use browser settings or locale overrides: some extensions (like GitHub Locale Changer) can also adjust date/time displays without coding.

For official feedback:

You can post a feature request directly in [GitHub Community → Feedback: Site Policy and Feature Requests](https://github.com/orgs/community/discussions/categories/feedback)

Or use GitHub Support → Share Feedback
.

Hopefully GitHub adds a native setting in the future, but this script-based workaround is currently the easiest and doesn’t require JavaScript knowledge.

💡 Why this is effective

It’s helpful, actionable, and beginner-friendly (good for “Accepted Answer”).

It includes external resource links (which users love).

It avoids repeating what’s already said and adds new value.

It uses GitHub’s helpful tone and Markdown formatting.

**Reply by Mishasama** [reply]
I can't find the "GitHub Date Format Converter" on GreasyFork. Please post the link here instead of just writing the name; thank you.


<details><summary>And please don't use AI's answer to fool us. @queenofcorgis </summary>
<p>

<img width="401" height="178" alt="Answer selected by queenofcorgis" src="https://github.com/user-attachments/assets/dd66b801-b00c-459c-bcdb-d877f776c6db" />

</p>
</details>

**Comment by spytheman** [comment]
The US format is confusing.

What is `9/10/2024` for example, by just looking at it?

Please use `2024/09/10` instead.

The ISO format has very logical reasons, and it is easy to understand to most people.

**Reply by ylluminate** [reply]
Even for some US natives the dates are confusing... ;) Much prefer the `YYYY-MM-DD` format.

**Reply by aral-matrix** [reply]
> Please use 2024/09/10 instead.

Bit late to the party, but no! If you're gonna mention the ISO format, don't propose something that's not conforming to the ISO 8601 format. The date separators are crucial to NOT mix up ISO8601 notation with that of any other, national, notation.
```YYYY-MM-DDThh:mm:ss.uuuuuu```
with the T or a blankspace being allowed, and an optional timezone following, is the only format to use. Obviously down to the desired accuracy.

**Comment by shahbaztube12** [comment]
Use Browser Extensions: Install extensions like Tampermonkey or Greasemonkey to run custom scripts that convert dates into your preferred format. You can also use libraries like Moment.js within these scripts to handle date and time formatting.

Time Zones and 24-Hour Format: Similar scripts can adjust the time display to show 24-hour format or convert times to your local timezone.

Submit a Feature Request: Consider suggesting this feature to GitHub through their feedback channels. If more users request this, GitHub might consider adding it as an option.

Conversion Tools: Use quick online date converters or scripts to manually convert GitHub dates into a format you’re comfortable with.

These steps can help you better manage date and time displays on GitHub.

and follow @technicalbaz on yt

**Comment by AKDub** [comment]
Just to add that using the US date format can also be tricky for native English speakers outside of the US. For example in this thread I thought the original post and reply were in August 2017, the next reply in August 2018, then after seeing August 2019 the cogs start turning and I realise it's all the same month - and that is after getting here directly from a 'how to change the date format on github' google search.

**Reply by aral-matrix** [reply]
> Just to add that using the US date format can also be tricky for native English speakers outside of the US.

Also, the DS date format (like many localized formats) sucks for anyone who needs their dates to sort alphabetically the same as chronologically, like.... for instance... programmers.

**Comment by pombredanne** [comment]
To add some insult to injury: See https://docs.github.com/en/issues/planning-and-tracking-with-projects/understanding-fields/about-date-fields

The date field in a project is displayed in US format but the tool demands to use ISO dates to filter, and if you paste a date field value (which uses US format) to use as a filter (which uses ISO dates) you will get for `targetdate:Nov 30, 2025`
> Filter contains 1 issue:
> Invalid value Nov for targetdate
> Fix by using date format YYYY-MM-DD 


Is there an issue reported for dates already?

**Comment by LucasImpulse** [comment]
The United States of America is not the centre of the world, even if GitHub is based out of it, there is an international standard even GitHub's date fields use. Given the site is developer oriented it makes sense to pursue ISO.
